managed/services/management/agent_test.go

```diff
@@ -89,13 +116,58 @@ func TestAgentService(t *testing.T) {
 			response, err := s.ListAgents(ctx, &agentv1beta1.ListAgentRequest{
 				ServiceId: service.ServiceID,
 			})
-
 			require.NoError(t, err)
-			assert.Len(t, response.Agents, 3) // 2 exporters + 1 agent
+
+			agents := []*agentv1beta1.UniversalAgent{
```

Comment:
Minor thing, but IMO it's better to call such variables `expected`. Then you're using it right where `expected` argument of assertion should be used. It'd be more readable.